src/Razor/src/Microsoft.AspNetCore.Razor.Language/Components/ComponentDocumentClassifierPass.cs

```diff
@@ -64,6 +64,14 @@ protected override CodeTarget CreateTarget(RazorCodeDocument codeDocument, Razor
                 computedClass = $"AspNetCore_{checksum}";
             }
 
+            var documentNode = codeDocument.GetDocumentIntermediateNode();
+            if (char.IsLower(computedClass, 0))
```

Comment:
I'd double check the implications of having the error logged at this layer. For instance at build time I believe you'll get two duplicate errors. one from the declaration generation pass and one from the actual razor compile.

It might make more sense for you to do this check at the TagHelperDescriptor provider layer. The descriptors have their own diagnostics which would also make sense to put this there.

Reply:
That doesn't happen when I test it locally. Also, we can't do it at the provider layer because we don't have access to the CodeDocument->SourceDocument->FilePath->FileName. I need that to provide this diagnostic. So, it doesn't make sense to put it there.

Reply:
If it doesn't happen then something else must be preventing it. Alright, can ignore this then 😄